src/main/java/bio/terra/service/dataset/flight/datadelete/DatasetDataDeleteFlight.java

```diff
@@ -0,0 +1,54 @@
+package bio.terra.service.dataset.flight.datadelete;
+
+import bio.terra.service.dataset.DatasetDao;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.dataset.flight.LockDatasetStep;
+import bio.terra.service.dataset.flight.UnlockDatasetStep;
+import bio.terra.service.iam.IamAction;
+import bio.terra.service.iam.IamResourceType;
+import bio.terra.service.iam.IamService;
+import bio.terra.service.iam.flight.VerifyAuthorizationStep;
+import bio.terra.service.job.JobMapKeys;
+import bio.terra.service.tabulardata.google.BigQueryPdao;
+import bio.terra.stairway.Flight;
+import bio.terra.stairway.FlightMap;
+import org.springframework.context.ApplicationContext;
+
+public class DatasetDataDeleteFlight extends Flight {
+
+    public DatasetDataDeleteFlight(FlightMap inputParameters, Object applicationContext) {
+        super(inputParameters, applicationContext);
+
+        // get the required daos and services to pass into the steps
+        ApplicationContext appContext = (ApplicationContext) applicationContext;
+        DatasetDao datasetDao = (DatasetDao) appContext.getBean("datasetDao");
+        DatasetService datasetService = (DatasetService) appContext.getBean("datasetService");
+        BigQueryPdao bigQueryPdao = (BigQueryPdao) appContext.getBean("bigQueryPdao");
+        IamService iamClient = (IamService) appContext.getBean("iamService");
+
+        // get data from inputs that steps need
+        String datasetId = inputParameters.get(JobMapKeys.DATASET_ID.getKeyName(), String.class);
+
+        addStep(new VerifyAuthorizationStep(
+            iamClient,
+            IamResourceType.DATASET,
+            datasetId,
+            IamAction.UPDATE_DATA));
+
+        // need to lock, need dataset name and flight id
+        addStep(new LockDatasetStep(datasetDao));
+
+        // check to see access to file and create external temp table
+        addStep(new CreateExternalTablesStep(bigQueryPdao, datasetService));
+
+        // insert into soft delete table
+        addStep(new DataDeletionStep(bigQueryPdao, datasetService));
+
+        // unlock
+        addStep(new UnlockDatasetStep(datasetDao));
+
+        // cleanup
+        addStep(new DropExternalTablesStep(bigQueryPdao, datasetService));
```

Comment:
A second run of soft deletes will collide on the names of external tables, if it starts before this completes.
I think either:
- Randomize the external table names in some way so they won't collide
- Reverse the order of this and the unlock.

Reply:
Soft deletes use the flight id for the suffix of the external table. This should prevent collisions, right?

Reply:
Yes, that'll do it.